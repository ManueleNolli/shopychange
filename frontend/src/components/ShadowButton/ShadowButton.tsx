import React from 'react'

import { Button, ButtonProps } from '@chakra-ui/react'
import BeatLoader from 'react-spinners/BeatLoader'

export type ShadowButtonProps = {
  label: string
  bgColor: string
  textColor: string
  shadowColor: string
  hoverColor: string
  focusColor: string
  onClick?: () => void
  isLoading?: boolean
  isDisabled?: boolean
  buttonProps?: ButtonProps
  type?: 'button' | 'submit' | 'reset'
}

export default function ShadowButton({
  label,
  bgColor,
  textColor,
  shadowColor,
  hoverColor,
  focusColor,
  onClick,
  isLoading,
  isDisabled = false,
  buttonProps,
  type = 'button',
}: ShadowButtonProps) {
  const shadowColor1 = `${shadowColor}7A`
  const shadowColor2 = `${shadowColor}6E`
  return isLoading ? (
    <Button
      data-testid="loading-button"
      isLoading
      w="6em"
      spinner={<BeatLoader size={8} color="white" />}
      bg={bgColor}
      color={textColor}
      boxShadow={`0px 1px 25px -5px ${shadowColor1}, 0 10px 10px -5px ${shadowColor2}`}
      _hover={{
        bg: hoverColor,
      }}
      _focus={{
        bg: focusColor,
      }}
      {...buttonProps}
    />
  ) : (
    <Button
      type={type}
      data-testid="onClick-button"
      isDisabled={isDisabled}
      onClick={onClick}
      bg={bgColor}
      style={{ whiteSpace: 'initial' }}
      color={textColor}
      boxShadow={`0px 1px 25px -5px ${shadowColor1}, 0 10px 10px -5px ${shadowColor2}`}
      _hover={{
        bg: hoverColor,
        cursor: isDisabled ? 'default' : 'pointer',
      }}
      _focus={{
        bg: focusColor,
      }}
      _disabled={{ bg: bgColor }}
      {...buttonProps}
    >
      {label}
    </Button>
  )
}
