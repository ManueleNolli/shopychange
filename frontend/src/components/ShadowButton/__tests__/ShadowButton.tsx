import React from 'react'
import { render } from '@testing-library/react'
import ShadowButton from '../ShadowButton'

describe('ShadowButton', () => {
  test('renders correctly on Loading', () => {
    const tree = render(
      <ShadowButton
        isLoading={true}
        label={''}
        bgColor={''}
        textColor={''}
        shadowColor={''}
        hoverColor={''}
        focusColor={''}
      />
    )
    expect(tree.getByTestId('loading-button')).toBeInTheDocument()
    expect(tree.queryByTestId('onClick-button')).not.toBeInTheDocument()
    expect(tree.container).toMatchSnapshot()
  })

  test('renders correctly on not Loading', () => {
    const tree = render(
      <ShadowButton
        label={''}
        bgColor={''}
        textColor={''}
        shadowColor={''}
        hoverColor={''}
        focusColor={''}
      />
    )
    expect(tree.getByTestId('onClick-button')).toBeInTheDocument()
    expect(tree.queryByTestId('loading-button')).not.toBeInTheDocument()
    expect(tree.container).toMatchSnapshot()
  })
})
