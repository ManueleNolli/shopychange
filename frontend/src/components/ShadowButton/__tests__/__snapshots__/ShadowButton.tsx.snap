// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShadowButton renders correctly on Loading 1`] = `
<div>
  <button
    class="chakra-button css-15109qs"
    data-loading=""
    data-testid="loading-button"
    disabled=""
    type="button"
  >
    <div
      class="chakra-button__spinner chakra-button__spinner--start css-1vsp78f"
    >
      <span
        style="display: inherit;"
      >
        <span
          style="display: inline-block; background-color: white; width: 8px; height: 8px; margin: 2px; border-radius: 100%; animation: react-spinners-BeatLoader-beat 0.7s 0s infinite linear; animation-fill-mode: both;"
        />
        <span
          style="display: inline-block; background-color: white; width: 8px; height: 8px; margin: 2px; border-radius: 100%; animation: react-spinners-BeatLoader-beat 0.7s 0.35s infinite linear; animation-fill-mode: both;"
        />
        <span
          style="display: inline-block; background-color: white; width: 8px; height: 8px; margin: 2px; border-radius: 100%; animation: react-spinners-BeatLoader-beat 0.7s 0s infinite linear; animation-fill-mode: both;"
        />
      </span>
    </div>
    <span
      class="css-q7lffx"
    />
  </button>
</div>
`;

exports[`ShadowButton renders correctly on not Loading 1`] = `
<div>
  <button
    class="chakra-button css-rjyonm"
    data-testid="onClick-button"
    style="white-space: initial;"
    type="button"
  />
</div>
`;
